ffmpeg/lpms_ffmpeg.c

```diff
@@ -847,6 +850,8 @@ static int open_input(input_params *params, struct input_ctx *ctx)
 
 open_input_err:
 fprintf(stderr, "Freeing input based on OPEN INPUT error\n");
+  avio_close(pb); // need to close manually, avformat_open_input
```

Comment:
Is the indentation here off?

Reply:
I see now that it's the fprintf which is indented wrong - feel free to fix that instead

Reply:
Fixed, waiting for the CI

Reply:
Also, just noticed that technically we could be passing in a null pointer here if `avformat_alloc_context` or `avio_open` fail, but it seems that `avio_close` already has a null check internally.

Defensively, I might do a `if (pb) avio_closep(&pb);` just to prevent that from ever blowing up in the future should anything change about avio's internal behavior, and to preemptively set `pb` to null. It's not strictly necessary here right now though; up to you.

Reply:
> Also, just noticed that technically we could be passing in a null pointer here if avformat_alloc_context or avio_open fail, but it seems that avio_close already has a null check internally.

I've checked this before committing - ffmpeg everywhere in the code relies on the ability to pass null pointer to `avid_close`. Say, `avformat_close_input` looks like this:
```c
void avformat_close_input(AVFormatContext **ps)
{
    AVFormatContext *s;
    AVIOContext *pb;

    if (!ps || !*ps)
        return;

    s  = *ps;
    pb = s->pb;

    if ((s->iformat && strcmp(s->iformat->name, "image2") && s->iformat->flags & AVFMT_NOFILE) ||
        (s->flags & AVFMT_FLAG_CUSTOM_IO))
        pb = NULL;

    flush_packet_queue(s);

    if (s->iformat)
        if (s->iformat->read_close)
            s->iformat->read_close(s);

    avformat_free_context(s);

    *ps = NULL;

    avio_close(pb);
}```

So I assumed that it is safe to pass null pointer there.